x-pack/legacy/plugins/maps/public/embeddable/types.ts

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+export interface MapFeature {
+  id: number;
+  layerId: string;
+}
+
+export interface LoadFeatureProps {
+  layerId: string;
+  featureId: number;
+}
+
+export interface FeatureProperty {
+  _propertyKey: string;
+  _rawValue: string | string[];
+}
+
+export interface FeatureGeometry {
+  coordinates: [number];
+  type: string;
+}
+
+export interface RenderTooltipContentParams {
+  addFilters(filter: object): void;
+  closeTooltip(): void;
+  features: MapFeature[];
+  isLocked: boolean;
+  getLayerName(layerId: string): Promise<string>;
+  loadFeatureProperties({ layerId, featureId }: LoadFeatureProps): Promise<FeatureProperty[]>;
```

Comment:
return type is `Promise<ITooltipProperty[]>`, https://github.com/elastic/kibana/blob/master/x-pack/legacy/plugins/maps/public/layers/tooltips/tooltip_property.ts#L10

Reply:
This change is causing some big repercussions. Doesn't seem to match what Siem is passing in.  I think I will switch back to `unknown` for this PR.

Reply:
<img width="778" alt="Screen Shot 2020-03-26 at 1 24 58 PM" src="https://user-images.githubusercontent.com/16563603/77676913-5b28c500-6f65-11ea-9a7b-fc1f3043219d.png">

Reply:
actually maybe if I do `TooltipProperty` it'll work, instead of `ITooltipProperty`

Reply:
How about just unknown for `loadFeatureProperties`? Might as well define as much as possible.

Reply:
I worked around it but had to use a cast to unknown in SIEM.

Reply:
This looks to be a by-product of SIEM accessing `_propertyKey` & `_rawValue` directly off of `TooltipProperty` (`getPropertyKey()` & `getRawValue()` interface wasn't around at the time). For this I think we can just update `featureProps` to be `ITooltipProperty` in `MapToolTipComponent`, `LineToolTipContent`, and `PointToolTipContent`, and then use `getPropertyKey()` & `getRawValue()` instead of accessing those props directly.

Feel free to leave the `unknown` @stacey-gammon and I can loop back to refactor after this PR is merged -- many thanks for helping clean things up here! 🙂 

Though one question for @nreese -- I had typed   `_rawValue` as `string | string[]`, but I see it's `string | undefined` in `ITooltipProperty`. Iirc I was indeed seeing `string[]` for fields like `host.ip` and such within the tooltip -- should `_rawValue` and `getRawValue()` include `string[]` as well?

Reply:
+1 for adding string[] to the value type